### DISCUSSION TITLE
Bulk Mapping Columns to Display Names

### DISCUSSION THREAD

**Discussion by tdt-amartin** [sqlalchemy/sqlalchemy#11470] [Usage Questions]
Hi All, I'm just now starting to think about how to solve this problem, is I don't have any examples. This is a thing that's starting to be painful, and I guess I don't know how to search for it or something because I feel certain this is something every web developer has had to deal with.

I have a data warehouse I've built with the typical dozens of sources flowing into it, and people want pretty display names in the UI. Not the database friendly names of columns. At this point there are thousands of columns in different places. And people a) do not always agree on what a specific column from a specific source should be named, and b) when they do agree, they often want to make changes.

I'd like to let users choose their own adventure and give them a UI to store display names and change them as they see fit and do it in a way where I don't have to change up any of my Mako templates. The storing part is simple enough if I start at the table level. Add a JSON column to each table that maps the column name to a display name.

Is there an idiomatic way to apply that map to a query result? Something around a custom .set_label_style()? The convenience function to take a query and a map and return a new query shouldn't be that hard, but I just can't help but think this kind of use case must be very common, and I'm missing something very obvious.

Edited to provide a clearer example:

Let's say I have two columns: development_team_priority_rank, and project_controls_priority_rank, and they want them to display in reports as Development Team Priority and Project Controls Critical Item Rank.

My result is a list of ORM objects UnifiedProjectMeta, and I've got a dictionary {"development_team_priority_rank": "Development Team Priority", "project_controls_priority_rank": "Project Controls Critical Item Rank"}

My question is whether there is anything baked into SQLAlchemy that allows you to do the mapping either at the query generation level by applying a labels or doing a mapping on the results?

I'm totally fine doing that myself on the results, I was just wondering.

**Comment by CaselIT** [comment]
Hi,

Not overly clear what you want to do, but assuming you have something like
- query builder
- get list of result
- display the list of result

in this case I would apply the display labels you want in the 3rd step.

**Reply by tdt-amartin** [reply]
Okay I edited to provide a concrete example. It sounds like the answer is roll your own, and that's totally cool I was just curious.

**Reply by CaselIT** [reply]
I don't think there is something automated, but i wonder if the results could be told to do that? Do you have a suggestion here @zzzeek ?

**Reply by tdt-amartin** [reply]
My very naive instinct is that it might be preferable (and I might try this approach before I do post processing) to pre-process the query and apply labels. That way the results are ready to go.


Post-processing in a general way I could apply to hundreds or thousands of tables might get . . . gross.

**Comment by tdt-amartin** [comment]
Actually I decided that I'm overthinking this by a lot. All I have to do is return the {column: display_name} map to the Mako template, and any place I need to display anything I can get the current display name by using the column name as the key in the dict.

I don't need to do any of this extra work anywhere.
